tests/test_sycl_queue.py

```diff
@@ -1418,6 +1418,18 @@ def test_take(func, device):
     assert_sycl_queue_equal(result_queue, expected_queue)
 
 
+@pytest.mark.parametrize(
+    "device",
+    valid_devices,
+    ids=[device.filter_string for device in valid_devices],
+)
+@pytest.mark.parametrize("sparse", [True, False], ids=["True", "False"])
+def test_indices(device, sparse):
+    dpnp_array = dpnp.indices((2, 3), sparse=sparse, device=device)
+    for i in dpnp_array:
+        assert i.sycl_device == device
```

Comment:
I'd suggest to make the check more detailed, like:
```suggestion
    sycl_queue = dpctl.SyclQueue(device)
    grid = dpnp.indices((2, 3), sparse=sparse, sycl_queue=sycl_queue)
    for dpnp_array in grid:
        assert_sycl_queue_equal(dpnp_array.sycl_queue, sycl_queue)
```